opcua/address_space.py

```diff
@@ -395,13 +395,17 @@ def set_attribute_value(self, nodeid, attr, value):
             attval = node.attributes[attr]
             old = attval.value
             attval.value = value
+            cbs = []
             if old.Value != value.Value:  # only send call callback when a value change has happend
-                for k, v in attval.datachange_callbacks.items():
-                    try:
-                        v(k, value)
-                    except Exception as ex:
-                        self.logger.exception("Error calling datachange callback %s, %s, %s", k, v, ex)
-            return ua.StatusCode()
+                cbs = list(attval.datachange_callbacks.items())
```

Comment:
I am not sure I understand. everything is under a lock so two thread cannot call that method at the same time. In addtion calling list() is not atomic in python.